File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/CandidateAdapter.java
package com.workforcex.employer.ui.employer;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.workforcex.employer.R;
import com.workforcex.employer.api.MatchedWorker;
import java.util.List;

public class CandidateAdapter extends RecyclerView.Adapter<CandidateAdapter.CandidateViewHolder> {

    private final List<MatchedWorker> workers;

    public CandidateAdapter(List<MatchedWorker> workers) {
        this.workers = workers;
    }

    @NonNull
    @Override
    public CandidateViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_candidate, parent, false);
        return new CandidateViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CandidateViewHolder holder, int position) {
        MatchedWorker w = workers.get(position);
        holder.tvRank.setText("#" + (position + 1));
        holder.tvName.setText(w.name != null ? w.name : w.mobileNumber);
        holder.tvScore.setText(String.format("Score: %.0f%%", w.score));
        holder.tvSkills.setText("Skills: " + (w.skills != null ? w.skills : "—"));
        holder.tvExperience.setText("Experience: " + (w.experience != null ? w.experience + " yrs" : "—"));
        holder.tvCity.setText("Location: " + (w.city != null ? w.city : "—"));
        holder.tvSalary.setText(w.preferredSalary != null ? "Expected: ₹" + w.preferredSalary.intValue() : "");
    }

    @Override
    public int getItemCount() { return workers.size(); }

    static class CandidateViewHolder extends RecyclerView.ViewHolder {
        TextView tvRank, tvName, tvScore, tvSkills, tvExperience, tvCity, tvSalary;

        CandidateViewHolder(View itemView) {
            super(itemView);
            tvRank = itemView.findViewById(R.id.tvRank);
            tvName = itemView.findViewById(R.id.tvName);
            tvScore = itemView.findViewById(R.id.tvScore);
            tvSkills = itemView.findViewById(R.id.tvSkills);
            tvExperience = itemView.findViewById(R.id.tvExperience);
            tvCity = itemView.findViewById(R.id.tvCity);
            tvSalary = itemView.findViewById(R.id.tvSalary);
        }
    }
}

File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/CandidateAdapter.kt
package com.workforcex.employer.ui.employer

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.workforcex.employer.R
import com.workforcex.shared_employer.models.MatchedWorker

class CandidateAdapter(private val workers: List<MatchedWorker>) :
    RecyclerView.Adapter<CandidateAdapter.CandidateViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CandidateViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_candidate, parent, false)
        return CandidateViewHolder(view)
    }

    override fun onBindViewHolder(holder: CandidateViewHolder, position: Int) {
        val w = workers[position]
        holder.tvRank.text = "#" + (position + 1)
        holder.tvName.text = w.name.ifBlank { w.mobileNumber }
        holder.tvScore.text = String.format("Score: %.0f%%", w.score)
        holder.tvSkills.text = "Skills: " + w.skills.ifBlank { "—" }
        holder.tvExperience.text = "Experience: " + w.experience + " yrs"
        holder.tvCity.text = "Location: " + w.city.ifBlank { "—" }
        holder.tvSalary.text = "Expected: ₹" + w.preferredSalary.toInt()
    }

    override fun getItemCount(): Int = workers.size

    class CandidateViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvRank: TextView = itemView.findViewById(R.id.tvRank)
        val tvName: TextView = itemView.findViewById(R.id.tvName)
        val tvScore: TextView = itemView.findViewById(R.id.tvScore)
        val tvSkills: TextView = itemView.findViewById(R.id.tvSkills)
        val tvExperience: TextView = itemView.findViewById(R.id.tvExperience)
        val tvCity: TextView = itemView.findViewById(R.id.tvCity)
        val tvSalary: TextView = itemView.findViewById(R.id.tvSalary)
    }
}
